internal/pkg/config/service.go

```diff
@@ -25,29 +25,29 @@ const (
 	AddonsCfnTemplateNameFormat = "%s.addons.stack.yml"
 )
 
-// Service represents a deployable long running service or task.
-type Service struct {
-	App  string `json:"app"`  // Name of the app this service belongs to.
-	Name string `json:"name"` // Name of the service, which must be unique within a app.
-	Type string `json:"type"` // Type of the service (ex: Load Balanced Web Server, etc)
+// Workload represents a deployable long running service or task.
+type Workload struct {
+	App  string `json:"app"`  // Name of the app this workload belongs to.
+	Name string `json:"name"` // Name of the workload, which must be unique within a app.
+	Type string `json:"type"` // Type of the workload (ex: Load Balanced Web Server, etc)
 }
 
-// CreateService instantiates a new service within an existing application. Skip if
-// the service already exists in the application.
-func (s *Store) CreateService(svc *Service) error {
-	if _, err := s.GetApplication(svc.App); err != nil {
+// CreateWorkload instantiates a new service or job within an existing application. Skip if
+// the workload already exists in the application.
+func (s *Store) CreateWorkload(wkld *Workload) error {
```

Comment:
💭 Mmm actually, I think we should keep the old signatures `GetService`, `CreateService`, ....
and instead have `getWorkload`, `deleteWorkload` to be private. 

The difference is that the workload methods, such as `getWorkload`, will just return the ssm error as is without wrapping. Then within `GetService` we can wrap the error so that the error messages reads `"get service %s from app %s: <ssm error message>"`

This way we have precise error message while still being DRY

Reply:
Fair, that's a good callout.